lib/assert.js

```diff
@@ -271,8 +271,14 @@ function innerDeepEqual(actual, expected, strict, memos) {
       position: 0
     };
   } else {
-    if (memos.actual.has(actual)) {
-      return memos.actual.get(actual) === memos.expected.get(expected);
+    // We prevent up to two set.has(x) calls by directly retrieving the value
+    // and checking for undefined.
```

Comment:
Sorry, my question is: what if `actual` or `expected` is the `undefined` value (I'm not sure it can be)? In that case, `map.has(actual)` is not the same as `map.get(actual) !== undefined`.